spirv_msl.cpp

```diff
@@ -5503,6 +5898,89 @@ void CompilerMSL::emit_instruction(const Instruction &instruction)
 		break;
 	}
 
+	case OpVectorTimesMatrix:
+	case OpMatrixTimesVector:
+	{
+		// If the matrix needs transpose, just flip the multiply order.
+		auto *e = maybe_get<SPIRExpression>(ops[opcode == OpMatrixTimesVector ? 2 : 3]);
+		if (e && e->need_transpose)
+		{
+			e->need_transpose = false;
+			string expr;
+
+			if (opcode == OpMatrixTimesVector)
+			{
+				expr = join(to_enclosed_unpacked_expression(ops[3]), " * ",
```

Comment:
invariant float math is not handled here.

Reply:
Also, no need to enclose expressions which go into functions, this is only used for binary ops like `*`, e.g. `1 + 2 * 3 + 4` might be fixed up to `(1 + 2) * (3 + 4)` but `foo(1 + 2, 3 + 4)` is fine.

Reply:
Removed the `enclose_expression` for one of the parameters as the other one uses `to_enclosed_unpacked_expression`.